.github/workflows/release-embedding-sdk.yml

```diff
@@ -0,0 +1,173 @@
+name: Release Metabase Embedding SDK for React
+
+on:
+  workflow_dispatch:
+    inputs:
+      sdk_version:
+        description: 'SDK version (e.g. 0.1.3)'
+        type: string
+        required: true
+      commit:
+        description: 'Optional full-length commit SHA-1 hash'
+
+concurrency:
+  # We want to ensure only one job is running at a time because
+  # there could be a conflict when updating the readme file.
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  git-tag:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 20
+    steps:
+      - name: Check out the code using the provided commit, or HEAD
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.commit }}
+
+      - name: Setup git user
+        run: |
+          git config --global user.email "metabase-bot@metabase.com"
+          git config --global user.name "Metabase bot"
+
+      - name: Create a new git tag
+        run: |
+          git tag embedding-sdk-${{ inputs.sdk_version }}
+
+      - name: Push the new tag
+        run: |
+          git push origin embedding-sdk-${{ inputs.sdk_version }}
+      - if: failure()
+        run: echo "Make sure the tag 'embedding-sdk-${{ inputs.sdk_version }}' doesn't exist."
+
+  test:
+    needs: git-tag
+    runs-on: ubuntu-22.04
+    timeout-minutes: 20
+    steps:
+      - name: Check out the code using the provided commit, or HEAD
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.commit }}
+
+      - name: Prepare front-end environment
+        uses: ./.github/actions/prepare-frontend
+
+      - name: Prepare back-end environment
+        uses: ./.github/actions/prepare-backend
+        with:
+          m2-cache-key: "cljs"
+
+      - name: Run unit tests
+        run: yarn embedding-sdk:test-unit
+
+  build-sdk:
+    needs: test
+    runs-on: ubuntu-22.04
+    timeout-minutes: 20
+    steps:
+      - name: Check out the code using the provided commit, or HEAD
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.commit }}
+
+      - name: Prepare front-end environment
+        uses: ./.github/actions/prepare-frontend
+
+      - name: Prepare back-end environment
+        uses: ./.github/actions/prepare-backend
+        with:
+          m2-cache-key: "cljs"
```

Comment:
nit: if this is the same as above (`test`), I wonder if we can cache them

Reply:
Good point, maybe we could bundle it. Although, the workflow would look cleaner since each job is responsible for certain things.
![image](https://github.com/metabase/metabase/assets/1937582/7ca96214-d904-4420-889e-81df70545253)

Edited: It seems our [FE](https://github.com/metabase/metabase/blob/master/.github/actions/prepare-backend/action.yml) and [BE](https://github.com/metabase/metabase/blob/master/.github/actions/prepare-backend/action.yml) preparation are already using cache for some steps, so they're not too slow. I don't think this should be a concern.

Also, I deliberately have `test` separated here because it's a sanity check if we could run the next 2 jobs in parallel (see the above workflow visualization) if we bundle `build-sdk` with `test` we probably have to do the same with `build-jar`

![image](https://github.com/metabase/metabase/assets/1937582/a5a20735-e933-41ff-a82d-9b7bc31a77bf)